internal/auth/yaml_storage.go

```diff
@@ -47,10 +59,25 @@ func (y *YamlStorage) WriteCalendarAuth(newCal CalendarAuth) (bool, error) {
 		return false, err
 	}
 
+	// Adding freshly written CalendarAuth to memory
+	// Probably unneeded, the next time this data will be retrieved is on the next calendarsync run
+	log.Debugf("Adding calendar auth for cal %s to memory", newCal.CalendarID)
+	y.DecryptedAuth = append(y.DecryptedAuth, newCal)
+
 	return true, nil
 }
 
 func (y *YamlStorage) ReadCalendarAuth(calendarID string) (*CalendarAuth, error) {
+	// if we already decrypted the file, read from memory
+	if len(y.DecryptedAuth) > 0 {
```

Comment:
There's no need to duplicate the calendarID filtering:
```
var calendars []CalendarAuth
if len(y.DecryptedAuth) > 0 {
    calendars = y.DecryptedAuth
} else {
    // readAndParseFile
}
for _, cal := range calendars {
```

Reply:
fixed :)